common/src/main/java/org/apache/uniffle/common/ShuffleDataResult.java

```diff
@@ -71,6 +71,16 @@ public byte[] getData() {
     return ByteBufUtils.readBytes(buffer.byteBuf());
   }
 
+  public int getDataLength() {
+    if (buffer == null) {
+      return 0;
+    }
+    if (buffer.nioByteBuffer().hasArray()) {
+      return buffer.nioByteBuffer().array().length;
+    }
+    return ByteBufUtils.readBytes(buffer.byteBuf()).length;
```

Comment:
This method `ByteBufUtils.readBytes` will cause memory copy.

Reply:
sorry, we should directly use `buffer.byteBuf()` array length